### PR TITLE
[AG-1904] Added an option to filter datasets in CLI

### PR DESCRIPTION
**Pull request by linglp**
# Problem

When testing or troubleshooting a specific dataset, there is no way to target it without modifying or copying the config file. These workarounds are error-prone — modified configs can be accidentally committed, introducing version errors in production configs.

# Solution
## Process a single dataset
```
adt configs/agora_preprod.yaml --dataset gene_info
```
## Process multiple datasets
```
adt configs/agora_preprod.yaml --dataset gene_info, team_info
```

# Test
* The command listed above are working as expected
* If you pass in: 
```
adt configs/agora_preprod.yaml --dataset "gene_info;team_info"
```
You would get an error: 
```
ValueError: No datasets found matching: ['gene_info;team_info']. Check that the dataset names match those in the config. To specify multiple datasets, use commas separated list
```

### CODE REVIEW

File: src/agoradatatools/process.py [hunk 1]
```diff
@@ -401,6 +403,14 @@ def process_all_files(
         run_id=run_id,
         table_id=gx_table,
     )
+    if filter_datasets:
+        datasets = [d for d in datasets if list(d.keys())[0] in filter_datasets]
```
Comment (jaclynbeck-sage) [inline src/agoradatatools/process.py:409]: Can you add a comment to make it more clear what list(d.keys())[0] is? It looks like the first (and only) key for d is the name of the dataset but it would help to have this in a comment.

File: src/agoradatatools/process.py [hunk 2]
```diff
@@ -401,6 +403,14 @@ def process_all_files(
...
+        if not datasets:
+            raise ValueError(
+                f"No datasets found matching: {filter_datasets}. "
+                "Check that the dataset names match those in the config. "
+                "To specify multiple datasets, use commas separated list"
```
Comment (jaclynbeck-sage) [inline src/agoradatatools/process.py]: nit: end of comment should probably be "use a comma-separated list"

File: tests/test_process.py [hunk 3]
```diff
@@ -1392,3 +1394,155 @@ def test_process_all_files_upload_false_process_dataset_fail(self, syn: Any):
             self.patch_load.assert_not_called()
             self.patch_format_link.assert_not_called()
             self.patch_update_table.assert_called_once()
+
```
Comment (jaclynbeck-sage) [inline tests/test_process.py:1397]: Can you add tests for if someone enters duplicate data sets, the data set should only be processed once?
i.e. from the function:
```
process_all_files(..., filter_datasets=["a", "a"])
```
or from the CLI:
```
adt ... --dataset gene_info --dataset gene_info
adt ... --dataset gene_info,gene_info
```

Your code in `process.py` looks robust to this case already, but I think it would be good to test for. I could see myself adding multiple --dataset tags and accidentally duplicating one!

**Review by jaclynbeck-sage — CHANGES_REQUESTED** [review]
This looks amazing! I only have very minor comments and requests for a few more test cases.

File: tests/test_process.py (hunk 3 above)
Comment (linglp) [reply tests/test_process.py:1397]: Thanks Jaclyn! I just added a test.

File: src/agoradatatools/process.py (hunk 1 above)
Comment (linglp) [reply src/agoradatatools/process.py:409]: Sure. I could add a comment!

File: tests/test_process.py (hunk 3 above)
Comment (linglp) [reply tests/test_process.py:1397]: Sure. See this test: `test_process_duplicated_dataset_names_filter_datasets`

**Review by jaclynbeck-sage — APPROVED** [review]
Awesome, looks great!
